src/Ocelot.Provider.Polly/PollyQoSProviderBase.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Ocelot.Configuration;
+using System.Net;
+
+namespace Ocelot.Provider.Polly;
+
+public class PollyQoSProviderBase
```

Comment:
I don't understand the meaning of this construct. Imo it should be an abstract class, and, since we have only static methods, why not use internal instead... I understand,  we shouldn't repeat ourselves, but I don't understand why an abstract/base class here.

Reply:
@ggnaegi 
> why not use **internal** instead...

We can't because of the [CS0060](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0060) rule of Microsoft C# conventions!
![image](https://github.com/ThreeMammals/Ocelot/assets/10501504/2d5a2a2f-f967-44bf-a04b-5422ff05f941)
So, we must keep it as `public` because descendant classes are public!

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1914/commits/a5bbabfdef7c710fb369f100f596956288365fd9 ✔️

Reply:
Raman, i'm fine with the class, why not have the methods as internal/public static, so you don't need to implement/extends and the polly providers implementations.

Reply:
If you're fine then resolve the issue plz!
I've already fixed it, see https://github.com/ThreeMammals/Ocelot/commit/a5bbabfdef7c710fb369f100f596956288365fd9!